src/wallet/fidelity.rs

```diff
@@ -409,6 +414,10 @@ impl Wallet {
             .fidelity_bond
             .insert(index, (bond, bond_spk, false));
 
+        log::info!("Syncing wallet");
+        self.sync()?;
+        log::info!("Sync successful");
```

Comment:
Ya thats true. But Its stll better to write data to file after major events. And not wait for shutdown only. Ex of major events are:
 - fidelity creation.
 - Swap completion.
 - Reovery completion.
 - Shutdown.
 
These are important data, and the maker can run for a long time without a shutdown in real-life. Relying on shutdown only write is risky.

Reply:
We should Ideally have a separate write thread for maker. Just like DNS. But I don't wanna do it right now because of possible unknown deadlock situation. Theres already a lot of threads accessing wallet data. Doing a write after major events is good enough for now.

Reply:
If it's small work then let's do it. If not then v0.1.1.  

Not a big problem, but good to fix it. In general security/funds critical data should not rely on memory.